### PR TITLE
Add koa mysql test app

### PR DESCRIPTION
**Pull request by luismiramirez**
## [Add Koa + MySQL test app](https://github.com/appsignal/appsignal-nodejs/commit/7cbc9845b35946c7323394d9dd09b401957944f8)

This test app uses Koa (web framework) and MySQL (database) it contains
4 endpoints:

- `GET /get`: Simple endpoint that always returns 200
- `GET /error`: Broken endpoint that throws an error
- `GET /mysql-query`: Performs a simple query using `mysql` client
  package
- `GET /mysql2-query`: Performs a simple query using `mysql2` client
  package

## [Add integration tests for Koa + MySQL test app](https://github.com/appsignal/appsignal-nodejs/commit/02b284fa46b199806635fd177ee759541d835efb)

Automated test suite for Koa + MySQL test app. It adds a couple of
useful helpers to get errors from span and SQL spans based on a parent
span name and instrumentation library.

## [Add waiting test methods to common helpers](https://github.com/appsignal/appsignal-nodejs/pull/719/commits/f711f51ef7e1ddcf0a81b97f3957a55b1e38d585) 

The `wait_for_start` and `clean_spans` methods are used by all the
integration test setups. This commit moves them to a common helper to
reduce duplication.

[skip changeset]

### CODE REVIEW

File: test/integration_helper.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
+# rubocop:disable Metrics/ModuleLength
 module IntegrationHelper
```
Comment (tombruijn) [inline test/integration_helper.rb:4]:
```suggestion
module IntegrationHelper # rubocop:disable Metrics/ModuleLength
```
Add it to the same line as the module definition and you don't need to disable it anymore at the end of the module definition.

File: test/integration_helper.rb [hunk 2]
```diff
@@ -111,4 +113,48 @@ def expect_ioredis_span(statement)
...
+        span["instrumentationLibrary"]["name"] == library
+    end
+    unless sql_span
+      raise "No SQL span with parent `#{span["parentSpanId"]}` and system `#{library}` found"
+    end
+
+    sql_span
+  end
 end
+# rubocop:enable Metrics/ModuleLength
```
Comment (tombruijn) [inline test/integration_helper.rb:160]:
```suggestion
```
See previous comment.

File: test/koa-mysql/app/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "koa-mysql-example",
+  "private": true,
+  "version": "1.0.0",
+  "main": "dist/app.js",
+  "scripts": {
+    "build": "tsc --version; tsc -p tsconfig.json",
```
Comment (tombruijn) [inline test/koa-mysql/app/package.json:7]:
```suggestion
    "build": "tsc -p tsconfig.json",
```
Is the version number being printed required?

File: test/koa-mysql/app/run.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (tombruijn) [inline test/koa-mysql/app/run.sh:1]:
```suggestion
#!/bin/bash

set -eu
```

Make sure it fails when a child process fails.

**Review by tombruijn — APPROVED** [review]

**Review by unflxw — APPROVED** [review]

File: test/koa-mysql/app/package.json (hunk 3 above)
Comment (luismiramirez) [reply test/koa-mysql/app/package.json:7]: It's not required per se, but I think it helps, knowing that a minimum version upgrade can break the build process.

File: test/koa-mysql/tests/spec/app_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  describe "GET /error" do
+    it "adds an error event in the Koa router span" do
+      response = HTTP.get("#{TEST_APP_URL}/error")
+      expect(response.status).to eq(500)
+
+      expect_http_root_span("GET /error")
+      expect_koa_router_span("/error")
+      expect_error_in_span(
```
[the remark this answers is not in the document]
Comment (luismiramirez) [reply test/koa-mysql/tests/spec/app_spec.rb:25]: I thought about this too. I will add this to the project board to work on a better helpers "framework" once we have the required setups.

**Pull request merged**